azure-transfer-wf/main.nf

```diff
@@ -115,36 +116,43 @@ workflow AzureTransferWf {
 
   main:
     study_id = permissibleStudy(study_id)
+    
+    if (params.enable_azure_songSub) {
+      DownloadMeta(
+        study_id,
+        Channel.from(analysis_id),
+        api_token
+      )
 
-    DownloadMeta(
-      study_id,
-      Channel.from(analysis_id),
-      api_token
-    )
-
-    Submit(
-      study_id,
-      DownloadMeta.out.payload_json,
-      api_token
-    )
+      Submit(
+        study_id,
+        DownloadMeta.out.payload_json,
+        api_token
+      )
```

Comment:
the `Submit` step is able to verify whether the song analysis with specified ID exists and good to use, if so, it will actually skip the actual submission. Does this not fullfil the use case you plan to handle? Is it the assumption that we don't need or want the existing song analysis object to be verified?